osmpbf/scanner.go

```diff
@@ -23,9 +23,10 @@ type Scanner struct {
 	// Skip element types that are not needed. The data is skipped
 	// at the encoded protobuf level, but each block still
 	// needs to be decompressed.
-	SkipNodes     bool
-	SkipWays      bool
-	SkipRelations bool
+	SkipNodes        bool
+	SkipWays         bool
+	SkipRelations    bool
+	SkipNodeFunction func(osm.Node) bool
```

Comment:
I think this is someone we should add. `NodeFilter(*osm.Node) bool` plus way and relation. If it returns false we reuse the thing passed in.

Reply:
Thanks for your review! Reusing the thing passing in if false might be a clever strategy to omit not needed allocs, not sure of the side effects.

Reply:
Sorry for dropping the ball . Introduced Filtering for Ways and Relations. Found no obvious way to reuse ways and nodes in case of the filter rejects it.